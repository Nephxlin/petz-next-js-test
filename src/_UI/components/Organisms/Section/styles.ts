import styled from "styled-components";

export const SectionContainer = styled.section`
  display: flex;
  flex-direction: column;
`

export const SectionTitle = styled.h1`
  font-size: 32px;
  font-weight: 700;
  margin: 12px 0;
`

export const SectionSubtitle = styled.p`
  font-size: 16px;
  font-weight: 400;
`