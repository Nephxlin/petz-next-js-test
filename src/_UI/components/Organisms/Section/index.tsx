import * as S from './styles'
import Container from "../../Layout/Container";
import Breadcrumbs from "../../Molecules/BreadCrumbs";

interface ISessionProps {
  title: string;
  subTitle: string;
  stylepattern?: 'primary' | 'white'
  breadCrumbs?: boolean;
}

const Section = ({ title = '', subTitle = '', stylepattern = 'primary' }: ISessionProps) => {
  return (
    <Container stylepattern={stylepattern}>
      <Breadcrumbs />
      <S.SectionContainer data-testid="section-component">
        <S.SectionTitle>{title}</S.SectionTitle>
        <p>{subTitle}</p>
      </S.SectionContainer>
    </Container>
  )
}

export default Section