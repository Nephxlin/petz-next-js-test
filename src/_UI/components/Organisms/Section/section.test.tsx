import 'jest-styled-components';
import { screen, render } from '@testing-library/react';
import Section from './index';
import mockRouter from 'next-router-mock';

jest.mock('next/router', () => jest.requireActual('next-router-mock'))

describe('Section component', () => {
  mockRouter.push("/initial-path");

  it('renders Section red', () => {
    const { container } = render(
      <Section
        stylepattern='primary'
        subTitle='its a long text'
        title='Hello World!'
      />
    );
    const section = screen.getByTestId('section-component')

    expect(section).toBeInTheDocument();
    expect(container.firstChild).toHaveAttribute('stylepattern', 'primary');
    expect(container.firstChild).toHaveStyleRule('background-color', '#E40F0F');
    expect(container).toMatchSnapshot();
  });

  it('renders Section white', () => {
    const { container } = render(
      <Section
        stylepattern='white'
        subTitle='its a long text'
        title='Hello World!'
      />
    );
    const section = screen.getByTestId('section-component')

    expect(section).toBeInTheDocument();
    expect(container.firstChild).toHaveAttribute('stylepattern', 'white');
    expect(container).toMatchSnapshot();
  });

});
