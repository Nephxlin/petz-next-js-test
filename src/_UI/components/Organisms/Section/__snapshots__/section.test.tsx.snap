// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section component renders Section red 1`] = `
.c5 {
  display: flex;
  flex-direction: column;
}

.c6 {
  font-size: 32px;
  font-weight: 700;
  margin: 12px 0;
}

.c0 {
  width: 100%;
  background-color: #E40F0F;
  color: #FFFFFF;
}

.c1 {
  display: flex;
  flex-direction: column;
  margin: 0 auto;
  max-width: 1440px;
  padding: 32px 106px;
  padding-bottom: 60px;
  width: 100%;
}

.c2 {
  display: flex;
  align-items: center;
  list-style: none;
  gap: 4px;
}

.c3 {
  display: flex;
  align-items: center;
  font-size: 12px;
  font-weight: 700;
  color: #ffffff;
  gap: 4px;
  text-decoration: none;
}

.c3:hover {
  text-decoration: underline;
}

.c4 {
  display: flex;
  align-items: center;
  font-size: 12px;
  font-weight: 700;
  color: #ffffff;
  gap: 4px;
  text-decoration: none;
}

.c4:before {
  content: ' > ';
  font-weight: 300;
}

<div>
  <div
    class="c0"
    data-testid="container-styled"
    stylepattern="primary"
  >
    <div
      class="c1"
    >
      <ul
        class="c2"
        data-testid="breadcrumbs-list"
      >
        <a
          href="/"
        >
          <li
            class="c3"
          >
            Home
          </li>
        </a>
        <li
          class="c4"
          data-testid="breadcrumbs-items"
        >
          Initial path
        </li>
      </ul>
      <section
        class="c5"
        data-testid="section-component"
      >
        <h1
          class="c6"
        >
          Hello World!
        </h1>
        <p>
          its a long text
        </p>
      </section>
    </div>
  </div>
</div>
`;

exports[`Section component renders Section white 1`] = `
.c5 {
  display: flex;
  flex-direction: column;
}

.c6 {
  font-size: 32px;
  font-weight: 700;
  margin: 12px 0;
}

.c0 {
  width: 100%;
}

.c1 {
  display: flex;
  flex-direction: column;
  margin: 0 auto;
  max-width: 1440px;
  padding: 32px 106px;
  padding-bottom: 60px;
  width: 100%;
}

.c2 {
  display: flex;
  align-items: center;
  list-style: none;
  gap: 4px;
}

.c3 {
  display: flex;
  align-items: center;
  font-size: 12px;
  font-weight: 700;
  color: #ffffff;
  gap: 4px;
  text-decoration: none;
}

.c3:hover {
  text-decoration: underline;
}

.c4 {
  display: flex;
  align-items: center;
  font-size: 12px;
  font-weight: 700;
  color: #ffffff;
  gap: 4px;
  text-decoration: none;
}

.c4:before {
  content: ' > ';
  font-weight: 300;
}

<div>
  <div
    class="c0"
    data-testid="container-styled"
    stylepattern="white"
  >
    <div
      class="c1"
    >
      <ul
        class="c2"
        data-testid="breadcrumbs-list"
      >
        <a
          href="/"
        >
          <li
            class="c3"
          >
            Home
          </li>
        </a>
        <li
          class="c4"
          data-testid="breadcrumbs-items"
        >
          Initial path
        </li>
      </ul>
      <section
        class="c5"
        data-testid="section-component"
      >
        <h1
          class="c6"
        >
          Hello World!
        </h1>
        <p>
          its a long text
        </p>
      </section>
    </div>
  </div>
</div>
`;
